### PR TITLE
[OneExplorer] Revise Refresh to re-build only the changed node

### PR DESCRIPTION
**Pull request by dayo09**
This commit revises refresh command and make it more efficient
to re-build only the changed node.

ONE-vscode-DCO-1.0-Signed-off-by: Dayoung Lee <dayoung.lee@samsung.com>

---

From #1402

### CODE REVIEW

File: src/OneExplorer/OneExplorer.ts [hunk 1]
```diff
@@ -505,9 +509,49 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<Node> {
...
+                case NodeType.baseModel:
+                  OneStorage.resetBaseModel(node.path);
+                  break;
+                case NodeType.config:
+                  OneStorage.resetConfig(node.path);
+                  break;
+                default:
+                  break;
+              }
+            }
```
Comment (dayo09) [inline src/OneExplorer/OneExplorer.ts:548]:
```suggestion
            };
```

**Comment by dayo09** [comment]
### Before
Refreshing all the nodes on deletion of any file.

1024-refresh-before

### After
Refreshing only the parent node of the deleted file.
1024-refresh

File: src/OneExplorer/OneStorage.ts [hunk 2]
```diff
@@ -162,4 +162,18 @@ export class OneStorage {
...
   }
+
+  public static resetBaseModel(path: string): void {
+    delete OneStorage.get()._baseModelToCfgsMap[path];
+    Logger.debug('OneStorage', `Base Mode Path(${path}) is removed.`);
+  }
+
+  public static resetConfig(path: string): void {
+    delete OneStorage.get()._cfgToCfgObjMap[path];
+    Object.entries(OneStorage.get()._baseModelToCfgsMap).forEach(([modelpath]) => {
```
Comment (llFreetimell) [inline src/OneExplorer/OneStorage.ts:173]: I thought following code is correct but not sure... @.@
Key of `_baseModelToCfgsMap` is array?

```suggestion
    Object.entries(OneStorage.get()._baseModelToCfgsMap).forEach((modelpath) => {
```

Comment (dayo09) [reply src/OneExplorer/OneStorage.ts:173]: Object.entries returns the given object's [key, value] list in an array.
For example,
[ ['model.tflite', 'model.cfg'],
  ['A.tflite', 'A.cfg']]

 will be `Object.entries(OneStorage.get()._baseModelToCfgsMap)`

I obtained the first value as modelpath within forEach statuement by destructuring the array. So it's the key.

**Review by llFreetimell — APPROVED** [review]
LGTM

**Comment by dayo09** [comment]
@Samsung/one-vscode PTAL :-D

**Review by mhs4670go — APPROVED** [review]
LGTM

**Pull request merged**